### PR TITLE
Feat/maintenance mode

### PR DESCRIPTION
**Pull request by Amean**
Adds a banner and message whenever app is in maintenance mode, maintenance mode is toggled by setting maintenanceMode to true in home.dart, https://github.com/Pylons-tech/pylons/issues/1295

### CODE REVIEW

File: wallet/lib/pages/home/home.dart [hunk 1]
```diff
@@ -285,6 +320,19 @@ class HomeScreenState extends State<HomeScreen> with SingleTickerProviderStateMi
                     width: 20.w,
                   )),
             ),
+            if (remoteConfigService.getMaintenanceMode())
+              Positioned(
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/home/home.dart:324]: Is it possible to create a widget called MaintenanceBanner and use it everywhere?

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Review by kjawadDeveloper2 — APPROVED** [review]

**Review by kamran8545 — APPROVED** [review]

**Review by afti-githobo — APPROVED** [review]

File: wallet/lib/components/maintenance_mode_widgets.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      color: AppColors.kDarkPurple,
+      padding: EdgeInsets.symmetric(vertical: 3.h, horizontal: 3.w),
+      child: Text(
+        LocaleKeys.maintenance_mode_header.tr(),
+        style: TextStyle(
+            color: Colors.white, fontSize: 10.sp, fontWeight: FontWeight.w700),
```
Comment (Kamran-Khan-Gene) [inline wallet/lib/components/maintenance_mode_widgets.dart:20]: format this code.

File: wallet/lib/pages/home/home.dart [hunk 3]
```diff
@@ -171,6 +177,12 @@ class HomeScreenState extends State<HomeScreen> with SingleTickerProviderStateMi
...
+                  top: 0.16.sh,
+                  right: 0,
```
Comment (Kamran-Khan-Gene) [inline wallet/lib/pages/home/home.dart:183]: format this.

File: wallet/lib/components/maintenance_mode_widgets.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+import '../generated/locale_keys.g.dart';
+import '../utils/constants.dart';
```
Comment (ahmadrns) [inline wallet/lib/components/maintenance_mode_widgets.dart:6]: Use absolute path instead of relative one.

File: wallet/lib/pages/home/home.dart [hunk 5]
```diff
@@ -22,6 +23,7 @@ import 'package:pylons_wallet/utils/screen_responsive.dart';
 import 'package:pylons_wallet/utils/svg_util.dart';
 
 import '../../generated/locale_keys.g.dart';
+import '../../services/third_party_services/remote_config_service/remote_config_service.dart';
```
Comment (ahmadrns) [inline wallet/lib/pages/home/home.dart:26]: Use absolute path

File: wallet/lib/pages/home/home.dart [hunk 6]
```diff
@@ -171,6 +177,12 @@ class HomeScreenState extends State<HomeScreen> with SingleTickerProviderStateMi
                     width: 20.w,
                   )),
```
Comment (ahmadrns) [inline wallet/lib/pages/home/home.dart:178]: code formatting not done properly

File: wallet/lib/services/third_party_services/remote_config_service/remote_config_service.dart [hunk 7]
```diff
@@ -18,6 +18,9 @@ abstract class RemoteConfigService {
 
   /// This method returns the ios app version in the remote config
   String getIOSAppVersion();
+
+  /// This method returns whether the app is in maintenance mode or not
```
Comment (ahmadrns) [inline wallet/lib/services/third_party_services/remote_config_service/remote_config_service.dart:22]: Update documentation it should be like this
/// This method tells whether the app is in maintenance mode or not
/// Output: return [bool] true if in maintenance mode else will return false

File: wallet/lib/components/maintenance_mode_widgets.dart (hunk 4 above)
Comment (kjawadDeveloper2) [reply wallet/lib/components/maintenance_mode_widgets.dart:6]: Good suggestion i will add a linter for this.

Comment (ahmadrns) [reply wallet/lib/components/maintenance_mode_widgets.dart:6]: Thank you.

**Pull request merged**